compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -554,7 +562,9 @@ impl<'hir> Map<'hir> {
     /// }
     /// ```
     pub fn get_fn_id_for_return_block(self, id: HirId) -> Option<HirId> {
-        let enclosing_body_owner = self.tcx.local_def_id_to_hir_id(self.enclosing_body_owner(id));
+        // Id may not have a owner if it's a fn itself
```

Comment:
Remark (not-blocking, just thinking): I almost wonder if this indicates a logic problem in whichever analysis requires `get_fn_id_for_return_block`.

Reply:
https://github.com/rust-lang/rust/blob/23f762d3b15eb897b1fd3199e8c075393b067b68/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs#L898-L906

I see this function calls `get_fn_id_for_return_block` directly, but I guess from the name that this function should handle fn HirId inputs. Should I add an extra check here and keep `get_fn_id_for_return_block` unchanged instead?

Reply:
I don't have a super strong opinion on this myself, I would wait for @fmease to discuss what we feel like these helpers should or should not be doing.